src/context/src/lib.rs

```diff
@@ -29,33 +29,56 @@ extern crate glutin;
 use amethyst_config::Element;
 use std::path::Path;
 
-mod video_context;
-pub use video_context::{VideoContext, DisplayConfig};
+pub mod video_context;
+pub mod broadcaster;
+pub mod event;
+use video_context::{VideoContext, DisplayConfig};
+use broadcaster::Broadcaster;
+use event::EngineEvent;
 
 config!(
     /// Contains configs for resources provided by `Context`
     struct Config {
     pub display_config: DisplayConfig = DisplayConfig::default(),
 });
 
-/// Contains all engine resources which must be shared by multiple parties, in particular `VideoContext`
+/// Contains all engine resources which must be shared by multiple parties, in particular `VideoContext` and `Broadcaster`
 pub struct Context {
     pub video_context: VideoContext,
+    pub broadcaster: Broadcaster,
 }
 
 impl Context {
-    /// Creates a `Context` configured according to `Config`
-    pub fn new(config: Config) -> Option<Context> {
-        let video_context =
-            match VideoContext::new(config.display_config) {
-                Some(video_context) => video_context,
-                None => return None,
-            };
-
-        Some(
-            Context {
-                video_context: video_context,
-            }
-        )
+    /// Create a `Context` configured according to `Config`
+    pub fn new(config: Config) -> Context {
+        let video_context = VideoContext::new(config.display_config);
+        let mut broadcaster = Broadcaster::new();
+        broadcaster.register::<EngineEvent>();
+
+        Context {
+            video_context: video_context,
+            broadcaster: broadcaster,
+        }
+    }
+
+    /// Return a vector containing all engine events
+    /// that have occured since the last call of this method
+    pub fn poll_engine_events(&mut self) -> Vec<EngineEvent> {
+        let mut events = vec!();
+        match self.video_context {
+            VideoContext::OpenGL { ref window, .. } =>
+                for event in window.poll_events() {
+                    let event = EngineEvent::new(event);
+                    events.push(event);
+                },
+            #[cfg(windows)]
+            VideoContext::Direct3D {  } => {
+                // stub
+                let event = EngineEvent::new(Event::Closed);
```

Comment:
It is guarded with #[cfg(windows)], so that is probably why it is accepted by Travis, and why I missed it (I built it only on linux).